### ISSUE TITLE
AWS CLI — ModuleNotFoundError: No module named 'awscli'

### ISSUE THREAD

**Issue by kevinmichaelchen** [pkgxdev/pantry#3503]
Moving the conversation from [here](https://github.com/pkgxdev/pantry/commit/a89611b335353d9c3ff7bf4b9c27a22ca456e091) for discoverability.

The temporary fix is just to manually constrain Python versions to less than 3.12:
```
pkgx +aws@latest +python.org@3.11 aws --version
pkgx install +python.org@3.11 aws
```

After that, subsequent invocations should work fine:
```
$ aws --version

aws-cli/2.13.22 Python/3.11.6 Darwin/22.6.0 source/arm64 prompt/off
```

**Comment by jhheider** [comment]
that should be "fixed" by a89611b335353d9c3ff7bf4b9c27a22ca456e091

**Comment by kevinmichaelchen** [comment]
i left one more comment in that thread.

i think the build failed

```
$ rm -rf ~/.pkgx/aws.amazon.com ~/.pkgx/python.org
$ pkgx aws --version
Traceback (most recent call last):
  File "/Users/kevin.c/.pkgx/aws.amazon.com/cli/v2.13.22/venv/bin/aws", line 19, in <module>
    import awscli.clidriver
ModuleNotFoundError: No module named 'awscli'
```

**Comment by mxcl** [comment]
We were foolish to dep to `python^3.11` rather than `python~3.11`

Jacob: what you think about a find replace for that?

**Comment by kevinmichaelchen** [comment]
yup, that did the trick, thanks!

```
$ rm -rf ~/.pkgx/aws.amazon.com ~/.pkgx/python.org
$ pkgx --sync

$ pkgx aws --version
aws-cli/2.13.22 Python/3.11.6 Darwin/22.6.0 source/arm64 prompt/off
```

**Comment by jhheider** [comment]
> Jacob: what you think about a find replace for that?

it's even worse than that. lots of stuff is ">=3.7". we had the same problem when llvm^15 released. i _think_ individual fixes, as needed, is best, but i can do something more global if desired.

**Comment by mxcl** [comment]
maybe we need a “don’t install python 3.12 *as a dependency* yet” flag.

Some deps are too dodgy to yolo onto users

**Comment by jhheider** [comment]
we _do_ sometimes see this with python, rust, llvm, go, and i can't think of others but they exist i'm sure. compatibility lag. solving it in the general case is... frightening? don't use a dep that under 2 weeks old unless explicit?

**Comment by mxcl** [comment]
> don't use a dep that under 2 weeks old unless explicit?

this is the right line of thinking.

**Comment by jhheider** [comment]
resolved by switching to python<3.12.
